### PR TITLE
Cleaning string_view

### PR DESCRIPTION
**Pull request by lemire**
This is just some trimming to make the code cleaner and more maintainable.  And maybe faster too though this is less certain.

### CODE REVIEW

File: include/simdjson/document.h [hunk 1]
```diff
@@ -219,7 +221,22 @@ class object : protected internal::tape_ref {
      * Get the key of this key/value pair.
      */
     inline std::string_view key() const noexcept;
-
+    /**
+    * Get the length (in bytes) of the key in this key/value pair.
```
Comment (jkeiser) [inline include/simdjson/document.h:225]: Nit: alignment of *

File: include/simdjson/document.h [hunk 2]
```diff
@@ -219,7 +221,22 @@ class object : protected internal::tape_ref {
...
+    * You should expect this function to be faster than key().size().
+    */
+    inline uint32_t key_length() const noexcept;
+    /**
+    * Returns true of the the key in this key/value pair is equal
```
Comment (jkeiser) [inline include/simdjson/document.h:230]: s/of/if in this and the next one

File: include/simdjson/document.h (hunk 1 above)
Comment (lemire) [reply include/simdjson/document.h]: Fixed.

File: include/simdjson/document.h (hunk 2 above)
Comment (lemire) [reply include/simdjson/document.h]: Fixed.

**Pull request merged**